### PR TITLE
Fix for the issue #24547 Magento\Customer\Model\Account\Redirect::setRedirectCookie() not properly working

### PR DESCRIPTION
**Pull request by sashas777**
### Description (*)
Added metadata to the setRedirectCookie and clearRedirectCookie functions.

### Fixed Issues (if relevant)
 
1. magento/magento2#24547: Magento\Customer\Model\Account\Redirect::setRedirectCookie() not properly working

### Manual testing scenarios (*)
1. Call method Magento\Customer\Model\Account\Redirect::setRedirectCookie()
 
### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by sashas777** [comment]
@paliarush  can you advise how to avoid these errors, please? 

```PHP Code Mess has found error(s):
/var/www/html/app/code/Magento/Customer/Model/Account/Redirect.php:28	The class Magento\Customer\Model\Account\Redirect uses sessions or cookies while not being a part of HTML Presentation layer
/var/www/html/app/code/Magento/Customer/Model/Account/Redirect.php:103	The method __construct has 10 parameters. Consider reducing the number of parameters to less than 10.
/var/www/html/app/code/Magento/Customer/Model/RedirectCookieManager.php:14	The class Magento\Customer\Model\RedirectCookieManager uses sessions or cookies while not being a part of HTML Presentation layer

Failed asserting that 2 matches expected 0.```

**Comment by drpayyne** [comment]
Hello @sashas777 
> The method __construct has 10 parameters. Consider reducing the number of parameters to less than 10

is common and can be suppressed by adding `@SuppressWarnings(PHPMD.ExcessiveParameterList)` to the method's PHPDoc Blocks.
> The class Magento\Customer\Model\RedirectCookieManager uses sessions or cookies while not being a part of HTML Presentation layer

can be suppressed by adding `@SuppressWarnings(PHPMD.CookieAndSessionMisuse)` to both the classes' PHPDoc Blocks.

Although, changing the constructor signature isn't backward compatible. Please refer to [Backward compatible development](https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/) for your issues.

File: app/code/Magento/Customer/Model/RedirectCookieManager.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /**
+     * Save redirect route to cookie for case of successful login/registration
+     *
+     * @param string $route
+     * @param StoreInterface $store
+     * @return void
+     */
+    public function setRedirectCookie($route, StoreInterface $store)
+    {
+        $cookieMetadata = $this->cookieMetadataFactory->createPublicCookieMetadata()
```
Comment (odubovyk) [inline app/code/Magento/Customer/Model/RedirectCookieManager.php]: Public cookie metadata is used. What will be the behavior and is it appropriate in the case if connection is secured?

**Comment by dyushkin** [comment]
Hello @sashas777!
Thank you for your contribution.
Methods `getCookieManager, setCookieManager, getRedirectCookie, setRedirectCookie, clearRedirectCookie` provided in `Magento\Customer\Model\Account\Redirect` class looks very legacy, and created with one purpose - to pass `login_redirect` cookie from `Magento/Checkout/view/frontend/web/js/sidebar.js` to backend for processing, and creating a redirect url.
Considering this, it is not much of use cases for these public methods, to create the implementation with its own single class. Can you please tag public methods I mentioned before as deprecated with the proper comment, and remove the `Magento/Customer/Model/RedirectCookieManager`. Please recheck suppress warnings are necessary.
Thank you,

**Comment by sashas777** [comment]
@dyushkin Updated. Does it look correct now?

**Comment by dyushkin** [comment]
@sashas777, thank you for quick updates, looks fine for me.

**Review by dyushkin — APPROVED** [review]

**Comment by engcom-Golf** [comment]
@magento run all tests

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-6891](https://jira.corp.magento.com/browse/ENGCOM-6891) has been created to process this Pull Request

**Review by dyushkin — APPROVED** [review]
All changes are correct.

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Pull request merged**